### PR TITLE
Residual attention network

### PR DESCRIPTION
**Pull request by PistonY**

### CODE REVIEW

**Comment by mli** [comment]
Job PR-431-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-431/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-431-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-431/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
Please update `tests/unittests/test_model_zoo.py` for new models.

**Comment by mli** [comment]
Job PR-431-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-431/8/index.html 
Code coverage of this PR:  vs. Master:

**Comment by PistonY** [comment]
OK.

**Comment by hetong007** [comment]
For naming consistency please replace `residualattentionnet56_32input` with `cifar_residualattentionnet56`.

**Comment by mli** [comment]
Job PR-431-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-431/12/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
LGTM. @zhreshold any comments?

**Comment by zhreshold** [comment]
LGTM, we will train with ImageNet, should be quick.

**Comment by hetong007** [comment]
@PistonY Do you have training results and command to share on CIFAR and ImageNet?

I'll go ahead and train the model on ImageNet to test its accuracy and performance.

**Comment by PistonY** [comment]
Here are some suggest.
https://discuss.gluon.ai/t/topic/8087/35
Have you ever test it on cifar？

**Comment by PistonY** [comment]
This version with p.t.r.m control I have no time to test it.Please first train it with attention-92 on cifar 10.If result is ok then try imagenet.

**Comment by PistonY** [comment]
fix rebase error of the branch.

**Comment by hetong007** [comment]
Experiment with `cifar_residualattentionnet92` gets to 95.7% on CIFAR10. Will test on ImageNet in the weekend.

**Comment by PistonY** [comment]
CI system always raise a error a few days later ,should I ignore it?

**Comment by zhreshold** [comment]
Don't worry about the CI if it can pass once.

**Comment by PistonY** [comment]
Ok.

**Comment by hetong007** [comment]
Attention56 achieves (21.03, 5.47) on ImageNet with our training setting. Thanks for the contribution!

**Comment by PistonY** [comment]
I may won't add AttentionResNext.I tried on Cifar-10 with mix-up only achieved 96.9 top1 acc and network much more larger than before(~500M under batch_size 64).

**Comment by weitingyuk** [comment]
@hetong007 LGTM, Would you share your trained ImageNet model?  thx
